[WIP] add analyze method SLIC to generate superpixels from image

johnnychen94 opened the pull request:
TODO:

- [x] enforce connectivity (#4)
- [ ] test cases
- [x] visualization

Reference:

[1] Radhakrishna Achanta, Appu Shaji, Kevin Smith, Aurelien Lucchi, Pascal Fua, and Sabine Süsstrunk, SLIC Superpixels, _EPFL Technical Report_ no. 149300, June 2010.

[2] The original [C++ codes](https://ivrl.epfl.ch/research-2/research-current/research-superpixels/)

[3] scikit-image's [`slic`](https://github.com/scikit-image/scikit-image/blob/master/skimage/segmentation/_slic.pyx)

johnnychen94 committed:
implement the main body of SLIC algorithm

johnnychen94 committed:
provide keyword constructor and override show for SLIC

Cuda-Chen commented:
Hi @johnnychen94,
after few months I finally implement enforce connectivity features.
The implementation is based on [scikit-image](https://github.com/scikit-image/scikit-image/blob/7e4840bd9439d1dfb6beaf549998452c99f97fdd/skimage/segmentation/_slic.pyx#L240-L348).

If you are pleased, I will send a PR

johnnychen94 commented:
This is so great a news to me! Please do so!

Cuda-Chen committed:
add enforce connectivity feature in analyze method (#4)

Add enforce connectivity feature in analyze method.
The enforce connectivity implementation is based on [scikit-image](https://github.com/scikit-image/scikit-image/blob/7e4840bd9439d1dfb6beaf549998452c99f97fdd/skimage/segmentation/_slic.pyx#L240-L348).

Because the original implementation uses 0-indexing, so I tried to translate into 1-indexing.
Though so far there are no any problems about `unbounded error`, it is ought to conduct test because I think
the implementation still have some problems in BFS part.
